### PR TITLE
feat(growth): filter OAuth/SSO redirect referrers from attribution

### PR DESCRIPTION
**Pull request by seanoliver**
## Problem

GitHub OAuth redirects and Google SSO set the browser's Referer header to their domain when redirecting back to supabase.com. Our attribution pipeline treats these as genuine referral traffic, inflating the `github` channel by ~20K orgs/week. The internal referrer fix (GROWTH-647) surfaced this by reducing `unknown-internal` — it didn't cause the issue, it revealed OAuth noise that was previously hidden.

## What happened

When users sign in with GitHub, the browser sends `Referer: https://github.com/`. GitHub's login pages use `origin-when-cross-origin` Referrer-Policy, which strips the path. So OAuth redirects arrive as bare `github.com/` — indistinguishable from a direct visit to github.com. Meanwhile, genuine GitHub referrals from repos/READMEs always include the full path because those pages use `no-referrer-when-downgrade`.

We validated against `mart_marketing_organization_attribution`: 98.5% of GitHub-attributed orgs have bare `github.com/` as the referrer. Only ~250/week have specific paths (genuine referrals).

## Changes

- Added `isOAuthRedirectReferrer()` to `first-referrer-cookie.ts` — identifies auth provider redirects:
  - `accounts.google.com` blocked entirely (dedicated SSO subdomain)
  - Bare `github.com/` blocked (OAuth redirect signature)
  - `github.com/<specific-path>` preserved (genuine repo/README referrals)
- Wired into `shouldRefreshCookie()` so OAuth referrers never get stamped into cookies
- Wired into `handlePageTelemetry()` referrer overrides as defense-in-depth
- 17 new tests covering all OAuth patterns and edge cases

## Testing

All 52 tests pass. New tests cover Google SSO (bare + with path), GitHub bare domain (with/without trailing slash), genuine GitHub referrals (repo, README, discussion, blob), explicit OAuth path, non-OAuth domains, empty/malformed URLs. Verified TDD — tests failed red before implementation, green after.

Companion dbt PR in data-engineering handles historical data.

GROWTH-732

### CODE REVIEW

File: packages/common/telemetry.tsx [hunk 1]
```diff
@@ -145,6 +149,7 @@ function handlePageTelemetry({
     if (
       firstReferrerData &&
       isExternalReferrer(firstReferrerData.referrer) &&
+      !isOAuthRedirectReferrer(firstReferrerData.referrer) &&
       !isExternalReferrer(pageData.ph.referrer)
```
[the remark this answers is not in the document]
Comment (seanoliver) [reply packages/common/telemetry.tsx]: Good catch — fixed in 645fdde. OAuth redirects are now treated as non-attribution referrers for override purposes, so the cookie's genuine referrer can win over a live OAuth redirect.

File: packages/common/telemetry.tsx [hunk 2]
```diff
@@ -123,7 +124,10 @@ function handlePageTelemetry({
     const storedReferrer = telemetryDataOverride?.ph?.referrer
 
     const shouldUseStoredReferrer = Boolean(
-      storedReferrer && isExternalReferrer(storedReferrer) && !isExternalReferrer(liveReferrer)
+      storedReferrer &&
+        isExternalReferrer(storedReferrer) &&
+        !isOAuthRedirectReferrer(storedReferrer) &&
+        !isExternalReferrer(liveReferrer)
```
[the remark this answers is not in the document]
Comment (seanoliver) [reply packages/common/telemetry.tsx]: Same fix applied here — 645fdde. The stored referrer can now override a live OAuth redirect.

File: packages/common/first-referrer-cookie.ts [hunk 3]
```diff
@@ -215,6 +253,7 @@ export function shouldRefreshCookie(
   request: { referrer: string; url: string }
 ): { stamp: boolean } {
   if (!existingCookie) {
+    if (isOAuthRedirectReferrer(request.referrer)) return { stamp: false }
```
[the remark this answers is not in the document]
Comment (seanoliver) [reply packages/common/first-referrer-cookie.ts:258]: Already addressed in ceb9f12 — the doc comment now includes the OAuth exclusion and documents why the paid-signal branch intentionally skips the OAuth check.

**Review by charislam — APPROVED** [review]
